ROOTFS/etc/waggle/sanity/warning/check_hanwha_camera.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+import sys
+import json
+import urllib.error, urllib.request
+import time
+
+def read_camera_manifest(path='/etc/waggle/node_manifest.json'):
+    with open(path, 'r') as file:
+        manifest = json.loads(file.read())
+        if 'cameras' not in manifest:
+            raise Exception(f'No cameras in the manifest: {path}')
+        else:
+            for camera in manifest['cameras']:
+                if camera['present'] == True:
+                    yield camera
+
+
+def http_request(url, user='waggle', pw='Waggle1@34', timeout=2):
+    passman = urllib.request.HTTPPasswordMgrWithDefaultRealm()
+    passman.add_password(None, url, user, pw)
+    authhandler = urllib.request.HTTPDigestAuthHandler(passman)
+    opener = urllib.request.build_opener(authhandler)
+    urllib.request.install_opener(opener)
+    headers = {'Accept': 'application/json'}
+    req = urllib.request.Request(url, headers=headers)
+    try:
+        with urllib.request.urlopen(req, timeout=timeout) as response:
+            return response.code, response.read()
+    except urllib.error.URLError:
+        return 404, b''
+
+
+def check():
+    code_when_exit = 0
+    cameras = list(read_camera_manifest())
+    # if no camera listed, we do not check
+    if len(cameras) == 0:
+        print('No cameras listed in the manifest. Exiting...')
+        exit(code_when_exit)
+
+    for n in range(10, 20, 1):
+        # get device info
+        return_code, body_json = http_request(f'http://10.31.81.{n}/stw-cgi/system.cgi?msubmenu=deviceinfo&action=view')
+        if return_code != 200:
+            continue
+        device_info = json.loads(body_json)
+
+        return_code, body_json = http_request(f'http://10.31.81.{n}/stw-cgi/video.cgi?msubmenu=snapshot&action=view')
+        if return_code != 200:
+            print(f'[ERROR] Failed to get a snapshot from 10.31.81.{n}', file=sys.stderr)
+            code_when_exit = 1
+
+        # check if the device model is listed in manifest
+        camera_to_delete = []
```

Comment:
why is this an array.  It seems that based on your logic below this would only be an array of size 1

Reply:
Right. Either using an array or an object (dict in this case) is fine here. However, an array type is easier for me to check if the variable has something, rather than checking like,
``` python3
camera_to_delete = None
# Do checking here ...
if camera_to_delete is not None:
```